pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -788,7 +788,8 @@ func (t *target) needsRemediation(timeoutForMachineToHaveNode time.Duration) (bo
 	}
 
 	// the node does not exist
-	if t.Node != nil && t.Node.UID == "" {
+	if t.Node != nil && t.Node.UID == "" && t.Node.Name == "" {
```

Comment:
This will never be true:

- The node is nil when noderef isn't set on the Machine yet.
- When noderef is set, and the Node was found, it has a UID.
- When the Node wasn´t found, the name is set by MHC code based on the noderef, but UID stays empty.

So we can never have a non nil node without name.

See https://github.com/openshift/machine-api-operator/blob/45519639494da10af8dbb5526be9e2d2311abcb4/pkg/controller/machinehealthcheck/machinehealthcheck_controller.go#L522